src/convert.js

```diff
@@ -67,7 +71,7 @@ function convertFeature(features, geojson, options) {
     } else if (type === 'GeometryCollection') {
         for (i = 0; i < geojson.geometry.geometries.length; i++) {
             convertFeature(features, {
-                id: geojson.id,
+                id: id,
```

Comment:
Same with geometry collections — the split parts won't get the same id with auto-id-generation.